### ISSUE TITLE
cube fails to build on 10.10.3+

### ISSUE THREAD

**Issue by bfontaine** [Homebrew/homebrew-science#2539]
```
==> Installing cube from homebrew/homebrew-science
==> Downloading http://apps.fz-juelich.de/scalasca/releases/cube/4.2/dist/cube-4.2.2.tar.gz
######################################################################## 100,0%
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/cube/4.2.2
==> make
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [cube] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

Logs: https://gist.github.com/d7a086c23020168e495e

**Comment by dpo** [comment]
```
libtool: link: g++ -I.../src/cube/include/ -pipe -O2 -arch x86_64 -Xarch_x86_64 -mmacosx-version-min=10.5 -Wall -W -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/local/Cellar/qt/4.8.7/mkspecs/macx-g++ -I. -I/usr/local/Cellar/qt/4.8.7/lib/QtCore.framework/Versions/4/Headers -I/usr/local/Cellar/qt/4.8.7/lib/QtCore.framework/Versions/4/Headers -I/usr/local/Cellar/qt/4.8.7/lib/QtNetwork.framework/Versions/4/Headers -I/usr/local/Cellar/qt/4.8.7/lib/QtNetwork.framework/Versions/4/Headers -I/usr/local/Cellar/qt/4.8.7/lib/QtGui.framework/Versions/4/Headers -I/usr/local/Cellar/qt/4.8.7/lib/QtGui.framework/Versions/4/Headers -I/usr/local/Cellar/qt/4.8.7/include -I. -I. -F/usr/local/Cellar/qt/4.8.7/lib -DCUBE_PREFIX=\"/usr/local/Cellar/cube/4.2.2\" -I./../src/cube/include -I./../src/cube/include/service -I./../src/cube/include/service/cubelayout -I./../src/cube/include/service/cubelayout/layout -I./../src/cube/include/service/cubelayout/readers -I./../src/cube/include/derivated -I./../src/cube/include/dimensions -I./../src/cube/include/dimensions/system -I./../src/cube/include/dimensions/calltree -I./../src/cube/include/dimensions/metric -I./../src/cube/include/dimensions/metric/strategies -I./../src/cube/include/dimensions/metric/data -I./../src/cube/include/dimensions/metric/cache -I./../src/cube/include/dimensions/metric/data/rows -I./../src/cube/include/dimensions/metric/index -I./../src/cube/include/dimensions/metric/matrix -I./../src/cube/include/dimensions/metric/value -I./../src/cube/include/dimensions/metric/value/trafo -I./../src/cube/include/dimensions/metric/value/trafo/single_value -I./../src/cube/include/syntax -I./../src/cube/include/syntax/cubepl/evaluators -I./../src/cube/include/syntax/cubepl/evaluators/nullary -I./../src/cube/include/syntax/cubepl/evaluators/unary -I./../src/cube/include/syntax/cubepl/evaluators/unary/functions/single_argument -I./../src/cube/include/syntax/cubepl/evaluators/binary -I./../src/cube/include/syntax/cubepl -I./../src/cube/include/topologies -I. -g -O2 -headerpad_max_install_names -arch x86_64 -Xarch_x86_64 -mmacosx-version-min=10.5 -o .libs/cube cube-Action.o cube-BoxPlot.o cube-Coloring.o cube-DownloadThread.o cube-FontWidget.o cube-Environment.o cube-HelpBrowser.o cube-HtmlWidget.o cube-MainWidget.o cube-MyTreeWidget.o cube-Plane.o cube-Point.o cube-PrecisionWidget.o cube-Pretty.o cube-Settings.o cube-SplitterDialog.o cube-Statistics.o cube-TableWidget.o cube-SystemTopologyWidget.o cube-SystemTopologyData.o cube-SystemTopologyDrawing.o cube-SystemTopologyToolBar.o cube-SystemTopologyViewTransform.o cube-TabWidget.o cube-Tetragon.o cube-TreeWidget.o cube-MetricTreeWidget.o cube-CallTreeWidget.o cube-SystemTreeWidget.o cube-FlatTreeWidget.o cube-TreeWidgetItem.o cube-TreeWidgetStatistics.o cube-TypedWidget.o cube-ValueWidget.o cube-MessageWidget.o cube-main.o cube-SystemBoxPlot.o cube-OrderWidget.o cube-TopologyDimensionBar.o cube-DimensionSelectionWidget.o cube-NewDerivatedMetricWidget.o cube-CubePLSyntaxHighlighter.o cube-CPPSyntaxHighlighter.o cube-FortranSyntaxHighlighter.o cube-VersionCheckWidget.o cube-HtmlHelpCollection.o cube-DerivedMetricsCollection.o cube-Action.moc.o cube-Coloring.moc.o cube-DownloadThread.moc.o cube-FontWidget.moc.o cube-HelpBrowser.moc.o cube-HtmlWidget.moc.o cube-MainWidget.moc.o cube-MessageWidget.moc.o cube-MyTreeWidget.moc.o cube-PrecisionWidget.moc.o cube-SplitterDialog.moc.o cube-SystemTopologyWidget.moc.o cube-SystemTopologyDrawing.moc.o cube-SystemTopologyToolBar.moc.o cube-SystemTopologyViewTransform.moc.o cube-SystemTopologyData.moc.o cube-SystemTopologyView.moc.o cube-DimensionSelectionWidget.moc.o cube-TabWidget.moc.o cube-TreeWidget.moc.o cube-SystemTreeWidget.moc.o cube-ValueWidget.moc.o cube-TableWidget.moc.o cube-TopologyDimensionBar.moc.o cube-OrderWidget.moc.o cube-NewDerivatedMetricWidget.moc.o cube-CubePLSyntaxHighlighter.moc.o cube-CPPSyntaxHighlighter.moc.o cube-FortranSyntaxHighlighter.moc.o cube-VersionCheckWidget.moc.o cube-resources.qrc.o -Wl,-F/usr/local/Cellar/qt/4.8.7/lib -Wl,-F/usr/local/Cellar/qt/4.8.7/lib -Wl,-bind_at_load  ./.libs/libcube4.dylib -L/usr/local/Cellar/qt/4.8.7/lib -framework QtGui -framework QtCore -framework QtNetwork -lz -ldbus-1
Undefined symbols for architecture x86_64:
  "cube::Cube::def_mirror(std::string const&)", referenced from:
      MainWidget::readFile(QString) in cube-MainWidget.o
  "cube::Cube::get_sev_adv(std::vector<std::pair<cube::Cnode*, cube::CalculationFlavour>, std::allocator<std::pair<cube::Cnode*, cube::CalculationFlavour> > >&, cube::Metric*, cube::CalculationFlavour)", referenced from:
      MetricTreeWidget::computeValue(std::vector<std::pair<cube::Cnode*, cube::CalculationFlavour>, std::allocator<std::pair<cube::Cnode*, cube::CalculationFlavour> > >&, TreeWidgetItem*) in cube-MetricTreeWidget.o
```
and lots more.

**Comment by apjanke** [comment]
Also broken on 10.11 for me.

Logs: https://gist.github.com/355174535d5cb2f221730779e21bbc86

**Comment by dpo** [comment]
This is now resolved, but see #4465 for the current failure (on 10.12).
